crates/cxx-qt-gen/src/naming/rust.rs

```diff
@@ -138,131 +129,52 @@ pub(crate) fn syn_type_is_cxx_bridge_unsafe(ty: &syn::Type) -> bool {
-        );
-    }
-
-    #[test]
-    fn test_syn_type_cxx_bridge_to_qualified_slice() {
-        let mappings = TypeNames::default();
-        assert_eq!(
-            syn_type_cxx_bridge_to_qualified(&parse_quote! { &[UniquePtr<T>] }, &mappings),
-            parse_quote! { &[cxx::UniquePtr<T>] }
-        );
-        assert_eq!(
-            syn_type_cxx_bridge_to_qualified(&parse_quote! { &mut [UniquePtr<T>] }, &mappings),
-            parse_quote! { &mut [cxx::UniquePtr<T>] }
-        );
-    }
-
-    #[test]
-    fn test_syn_type_cxx_bridge_to_qualified_tuple() {
-        assert_eq!(
-            syn_type_cxx_bridge_to_qualified(
-                &parse_quote! { (UniquePtr<T>, ) },
-                &TypeNames::default()
-            ),
-            parse_quote! { (cxx::UniquePtr<T>, ) }
-        );
+    macro_rules! test_bridge_types_qualified {
+        [$($input_type:tt => $output_type:tt),*] => {
+            let mut type_names = TypeNames::mock();
+            type_names.insert("T", None, None, None);
+
+            $(
+            let expected_type : syn::Type = parse_quote! $output_type ;
+            crate::tests::assert_tokens_eq(
+                &syn_type_cxx_bridge_to_qualified(&parse_quote! $input_type, &type_names).unwrap(),
+                expected_type.to_token_stream()
+            );
+            )*
+        }
     }
 
     #[test]
-    fn test_syn_type_cxx_bridge_to_qualified_mapped() {
-        let mut mappings = TypeNames::default();
-        mappings.insert("A", Some(format_ident!("ffi")), Some("B"), None);
-        assert_eq!(
-            syn_type_cxx_bridge_to_qualified(&parse_quote! { A }, &mappings),
-            parse_quote! { ffi::A }
-        );
+    fn test_syn_type_cxx_bridge_to_qualified() {
+        test_bridge_types_qualified! [
+        // Primitive types
+        { i32 } => { i32 },
+        { f32 } => { f32 },
+        { bool } => { bool },
+        { String } => { String },
+        // CXX implementation of C++ types
+        { CxxString } => { cxx::CxxString },
+        { CxxVector<T> } => { cxx::CxxVector<T> },
+        { SharedPtr<T> } => { cxx::SharedPtr<T> },
+        { UniquePtr<T> } => { cxx::UniquePtr<T> },
+        { WeakPtr<T> } => { cxx::WeakPtr<T> },
+        // Pin
+        { Pin<&mut T> } => { core::pin::Pin<&mut T> },
+        // Different nestings
+        { [UniquePtr<T>; 1] } => { [cxx::UniquePtr<T>; 1] },
+        { fn(UniquePtr<T>) -> SharedPtr<T> } => { fn(cxx::UniquePtr<T>) -> cxx::SharedPtr<T> },
+        { Pin<UniquePtr<T>> } =>  { core::pin::Pin<cxx::UniquePtr<T>> },
+        { *mut UniquePtr<T> } => { *mut cxx::UniquePtr<T> },
+        { &UniquePtr<*mut T> } => { &cxx::UniquePtr<*mut T> },
+        { &[UniquePtr<T>] } => { &[cxx::UniquePtr<T>] },
+        { &mut [UniquePtr<T>] } => { &mut[cxx::UniquePtr<T>] },
+        { (UniquePtr<T>, ) } => { (cxx::UniquePtr<T>, ) },
+        // Mapped type
+        { MyObject } => { qobject::MyObject }
```

Comment:
do we need to test any rust_name mappings ? eg where the cxx bridge type name is different to the qualified ?

Reply:
OK after conversation turns out we figured out a plan here and that the rust name is always used.